src/anyvar/storage/postgres.py

```diff
@@ -1,93 +1,227 @@
-            "where length(vrs_object -> 'state' ->> 'sequence') = 0"
-        )
-        return data[0]
+        :param name: key to delete object for
+        """
+        name = str(name)  # in case str-like
+        with self.conn.cursor() as cur:
+            cur.execute(
+                "DELETE FROM vrs_objects WHERE vrs_id = %s;",
+                [name]
+            )
+        self.conn.commit()
 
-    def substitution_count(self):
-        data = self.conn._fetchone(
-            "select count(*) as c from vrs_objects "
-            "where length(vrs_object -> 'state' ->> 'sequence') = 1"
-        )
-        return data[0]
+    def close(self):
+        """Terminate connection if necessary."""
+        if self.conn is not None:
+            self.conn.close()
 
-    def insertion_count(self):
-        data = self.conn._fetchone(
-            "select count(*) as c from vrs_objects "
-            "where length(vrs_object -> 'state' ->> 'sequence') > 1"
-        )
-        return data[0]
+    def __del__(self):
+        """Tear down DB instance."""
+        self.close()
+
+    def __len__(self):
+        with self.conn.cursor() as cur:
+            cur.execute("""
+                SELECT COUNT(*) AS c FROM vrs_objects
+                WHERE vrs_object ->> 'type' = 'Allele';
+            """)
+            result = cur.fetchone()
+        if result:
+            return result[0]
+        else:
+            return 0
+
+    def get_variation_count(self, variation_type: VariationStatisticType) -> int:
+        """Get total # of registered variations of requested type.
+
+        :param variation_type: variation type to check
+        :return: total count
+        """
+        if variation_type == VariationStatisticType.TEXT:
+            return self._text_count()
+        if variation_type == VariationStatisticType.SUBSTITUTION:
+            return self._substitution_count()
+        elif variation_type == VariationStatisticType.INSERTION:
+            return self._insertion_count()
+        elif variation_type == VariationStatisticType.DELETION:
+            return self._deletion_count()
+        else:
+            return self._substitution_count() + self._deletion_count() + \
+                self._insertion_count()
+
+    def _text_count(self) -> int:
+        """Get total # of registered text variations.
+
+        :return: total count
+        """
+        with self.conn.cursor() as cur:
+            query = """
+            SELECT COUNT(1) AS c FROM vrs_objects
+            WHERE vrs_object ->> 'type' = 'Text'
+            """
+            cur.execute(query)
+            result = cur.fetchone()
+        if result:
+            return result[0]
+        else:
+            return 0
+
+    def _deletion_count(self) -> int:
+        with self.conn.cursor() as cur:
+            cur.execute("""
+                select count(*) as c from vrs_objects
+                where length(vrs_object -> 'state' ->> 'sequence') = 0;
```

Comment:
I know this wasn't you, but we want to make a new issue for this. For example, `NC_000003.12:g.10146527_10146528delCT` returns

```json
{
    "_id": "ga4gh:VA.hvwBZON5KzQGQazIMpeUu_dmyJ-xN8EV",
    "type": "Allele",
    "location": {
      "_id": null,
      "type": "SequenceLocation",
      "sequence_id": "ga4gh:SQ.Zu7h9AggXxhTaGVsy7h_EZSChSZGcmgX",
      "interval": {
        "type": "SequenceInterval",
        "start": {
          "type": "Number",
          "value": 10146524
        },
        "end": {
          "type": "Number",
          "value": 10146528
        }
      }
    },
    "state": {
      "type": "LiteralSequenceExpression",
      "sequence": "CT"
    }
  }
```

This means we might want to look at what we have for the other summary methods too since fully-justified normalization may mess with them too.